### DISCUSSION TITLE
Admonitions in Goldmark?

### DISCUSSION THREAD

**Discussion by cdupont** [yuin/goldmark#174] [Ideas]
Some Markdown flavors supports "admonitions" (also named "call outs"): https://python-markdown.github.io/extensions/admonition/
They are useful for putting something forward in the flow of text.
For instance, using "!!!" as syntax for the admonition:

```
!!! warning
    Warning, this tutorial is in development.
```
Results in something like that:
![image](https://user-images.githubusercontent.com/484033/102905804-f6f34d00-4473-11eb-9986-56b9e0c9e0d4.png)

Several types of admonitions can be supported, such as "note", "tip", "warning" or "action".

Do you think this would be useful? I think it could be implemented as an extension (I could try, time permitting :).

**Comment by tomcam** [comment]
I would absolutely love it but it's not in the CommonMark spec nor in the other extensions already supported, such as the PHP and GitHub extensions.

**Reply by vanbroup** [reply]
> **Note**
> GitHub does now has limited support for Admonitions 

GitHub does now: https://github.com/community/community/discussions/16925

**Comment by u0nel** [comment]
hedge doc uses colons:
```
:::warning
The warning
:::
```
https://demo.hedgedoc.org/eqY3mmlXT_qTFMB9L1_rLg#

**Comment by jiriks74** [comment]
Github currently support the following syntax in all places where Markdown is present.

```md
> [!Note]
> This is a note
```

> [!Note]
> This is a note.

Others are possible:

> [!Tip]
> This is a tip!

> [!Warning]
> This is a warning!

> [!Important]
> This is important!

> [!Caution]
> Be cautious!

The same syntax is currently supported by Gitea too.

---

For more information see [this Github discussion](https://github.com/orgs/community/discussions/16925)

**Comment by cmpute** [comment]
Related: https://github.com/yuin/goldmark/issues/421 and https://github.com/yuin/goldmark/issues/422
